src/content/1.7/modules/concepts/controllers/admin-controllers/override-decorate-controller.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Override or decorate a Core controller
+---
+
+# Override or decorate a Core controller
+
+{{< minver v="1.7.7" title="true" >}}
+
+If you want to extend the behavior of a Back Office page in PrestaShop, you have multiple options.
+
+Most of standard extension needs can be fulfilled using one [hook][hooks].
+
+If there is no hook available for your need, and you only need to modify the display of the page, you might want to [override the template][template-override].
+
+But sometimes you want to modify the page deeper. In this case, you need to modify the controller behavior.
+
+If the Back Office page you want to modify is powered by Symfony, you have 3 options:
+
+- remap the route to target a new Controller you create
+- override the existing Core controller
+- decorate the existing Core controller
+
+## Remap the routing
+
+In Symfony, [routes](https://symfony.com/doc/3.4/routing.html) such as `/sell/orders/orders` are mapped to controllers by YAML configuration files such as this one:
+```yaml
+admin_orders_index:
+  path: /sell/orders/orders/
+  methods: [GET]
+  defaults:
+    _controller: PrestaShopBundle:Admin/Sell/Order/Order:index
+    _legacy_controller: AdminOrders
+    _legacy_link: AdminOrders
+```
+
+In your module, you have the possibility to replace this configuration item by your own.
+
+So for example if you want that people browsing the Back Office on URL `/sell/orders/orders/` hit your own controller `MyModule\Controller\DemoController`, you can do this:
+```yaml
+# modules/your-module/config/routes.yml
+admin_orders_index:
+    path: /demo/orders
```

Comment:
Shouldn't this be the same route? Otherwise it will only reroute internal links
```suggestion
    path: /sell/orders/orders/
```

Reply:
Yes, I wanted to make clear people _do not have to keep the same route_ and so _they are given an extra level of freedom_

However indeed it means external links will return HTTP 404 🤔 

I will add a tip instead

Reply:
Actually module routes are prefixed with `/modules` so we cannot have the exact same path as before when re-routing a Symfony route

Needs to be explored and fixed ... outside of the scope of this PR 😭